core/src/bls_sigverify/bls_sigverifier.rs

```diff
@@ -135,8 +138,22 @@ impl BLSSigVerifier {
 
             match message {
                 ConsensusMessage::Vote(vote_message) => {
-                    // Only need votes newer than root slot
-                    if vote_message.vote.slot() <= root_bank.slot() {
+                    // If node will be a leader in 8 slots, it needs to keep older vote messages otherwise, discard older messages to avoid unnecessary verification compute.
+                    let vote_slot = vote_message.vote.slot();
+                    let slot_to_check = match self
+                        .leader_schedule_cache
+                        .slot_leader_at(vote_slot + 8, None)
```

Comment:
The `8` should be defined as a const somewhere.  Not sure where is a good place though.  Suggestions from reviewers are welcome.